### ISSUE TITLE
Add helpers for other types of content

### ISSUE THREAD

**Issue by kohenkatz** [tomschlick/laravel-http2-server-push#11]
I can do a PR for this, but first I wanted to gauge interest.

The [`preload` spec](https://www.w3.org/TR/preload/) that defines server push lists the valid "request destination" types for the `as` attribute of the `Link` header. Of those, only `image`, `script`, and `style` have helpers in this package.  While those are likely the most commonly used, I think that we should make it easier for users to specify the other types.

There are two ways that I can think of to do this:
1. Add a generic helper `pushFile` that has a second argument which defines the type.
2. Add helpers for all of the types.

Here is the example list of types from the documentation (including the three types already supported):
- media - for `<audio>` and `<video>` tags
- script - for `<script>` tags and `importScripts` calls (inside workers)
- style - for `<link rel=stylesheet>` and `@import`
- font - for CSS `@font-face` - _this should also have a `type` attribute which holds the mime type of the font file_
- image - for `<img>`, `<picture>`, `srcset`, `imageset`, SVG's `<image>` tag, CSS `*-image`
- _none_ - for `XHR`, `fetch`
- worker - for `Worker`, `SharedWorker`
- embed - for `<embed>`
- object - for `<object>`
- document - for `<iframe>` and `<frame>`

The spec also defines the following types, but they are not listed in this part of the documentation:
- report
- serviceworker
- sharedworker
- manifest
- xslt

Thoughts?

**Comment by tomschlick** [comment]
Yeah this has been on my todo list since I created the package but I haven't found the time yet...

I'd be all for a PR if you have the time to do so!

Maybe we go for a generic `pushItem($path, $type)` method but keep the ones for image, style, script.

**Comment by kohenkatz** [comment]
Some time in the next few days I will try to get a PR in for a `pushItem` helper.

In addition to `$path`, and `$type`, there will also be a `$mimeType` argument for things like `font` to use. It will default to `null` because it isn't needed everywhere.

**Comment by tomschlick** [comment]
Sounds good. Thanks!

**Comment by regiszanandrea** [comment]
Hi everyone,

I created a PR that supports Fonts, could be useful for you

PR: https://github.com/tomschlick/laravel-http2-server-push/pull/19
